lib/layers/data/repositories/repository_impl.dart

```diff
@@ -0,0 +1,225 @@
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:logger/logger.dart';
+import 'package:wot_statistic/common/constants/constants.dart';
+import 'package:wot_statistic/layers/data/models/remote/achievements_data/achievement_data.dart';
+import 'package:wot_statistic/layers/data/models/remote/vehicles_data/vehicles_data_meta.dart';
+import 'package:wot_statistic/layers/data/models/remote/vehicles_data/vehicles_data_ttc.dart';
+import 'package:wot_statistic/layers/data/sources/local_data_source.dart';
+import 'package:wot_statistic/layers/domain/entities/personal_data.dart';
+import 'package:wot_statistic/layers/domain/entities/user.dart';
+import 'package:wot_statistic/layers/domain/entities/vehicles_data.dart';
+import 'package:wot_statistic/layers/domain/repositories/repository.dart';
+
+import '../../../common/constants/network_const.dart';
+import '../../domain/entities/achieves.dart';
+import '../models/local/user_data.dart';
+import '../models/remote/achievements_data/achievements_database.dart';
+import '../models/remote/clan_info/clan_info.dart';
+import '../models/remote/personal_api_data/personal_data_api.dart';
+import '../models/remote/user_achieves/user_achieves_api_data.dart';
+import '../models/remote/user_vehicles/user_vehicle.dart';
+import '../models/remote/user_vehicles/user_vehicles_api.dart';
+import '../models/remote/vehicles_data/vehicles_data.dart';
+import '../sources/remote_data_source.dart';
+
+class RepositoryImpl extends Repository {
```

Comment:
Interface segregation for Repository is broken, it would be goot to split it to different repositories.